server/src/service/module.ts

```diff
@@ -1,16 +1,75 @@
-import { Game, GameDocument } from "../model/game";
 import { ModuleDocument, Module } from "../model/module";
+import { ActivityInput, ActivityDocument, Activity } from "../model/activity";
+import { activityService } from "./activity";
 import { LinkedListService } from "./utils/linkedlist";
+import underscore from "underscore";
+import { ObjectNotFoundError } from "./errors";
 
-export default class ModuleService extends LinkedListService<ModuleDocument, GameDocument> {
+export default class ModuleService extends LinkedListService<ModuleDocument> {
   constructor() {
     super({
       model: Module,
       select: "name image imageAlt next",
-      metaModel: Game,
-      createMeta: true,
     });
   }
+
+  async findByActivity({activity, select}: {
+    activity: string; 
+    select?:string
+  }): Promise<ModuleDocument | null> {
+    return await Module.findOne({activies: {$in: [activity]}})
+      .select(select || "")
+      .exec();
```

Comment:
Qual a razão ter esse método?

Reply:
Pro caso de uma atividade ser deletada ela também ser removida do módulo que ela pertence. Se atividade A que está dentro do módulo X for deletada, ele usa o findByActivity para remover a atividade A de dentro de X. Isso é algo que roda pelos métodos do AdminJS.